crystal_grpc/Dockerfile

```
@@ -0,0 +1,16 @@
+FROM crystallang/crystal:latest
+
+WORKDIR /app
+COPY crystal_grpc /app
+COPY proto /app/proto
+
+RUN shards install
+RUN apt update && apt install -y protobuf-compiler protobuf-compiler-grpc
+RUN mkdir -p src/protobufs && protoc -I proto \
+    --grpc_out=src/protobufs \
+    --crystal_out=src/protobufs \
+    --plugin=protoc-gen-grpc=bin/grpc_crystal \
+    --plugin=protoc-gen-crystal=bin/protoc-gen-crystal \
+    proto/helloworld/helloworld.proto
+
+ENTRYPOINT crystal src/server.cr
```

Running it through `crystal` requires it to compile every time the container boots. This suggestion moves compilation to the container build phase so that the app is ready for connections as soon as the container starts up.

```suggestion
RUN crystal build src/server.cr -o bin/server

ENTRYPOINT bin/server
```

---

Thank you! This seems to have solved the problem. Though having the benchmark run for 30s I expected the compilation to end sooner and at least manage to serve some requests. At least on my machine the build command was quite fast.

Anyway, I added the `--release` flag to your suggestion, now it happily beats the pure Ruby performance! Rock solid 💪